### ISSUE TITLE
Use `gitutils` in the `GIT_COMMITS` syncer

### ISSUE THREAD

**Issue by patrickdevivo** [mergestat/mergestat#406]
Switch to using [`gitutils`](https://github.com/mergestat/gitutils) in the `GIT_COMMITS` syncer - if it makes sense to do so (it's not signficantly slower).

We probably want to wait on some benchmarks before we invest in the effort.

**Comment by ramiroJCB** [comment]
@patrickdevivo  Exactly what feature of gitutils we should use in GIT_COMMITS?

**Comment by patrickdevivo** [comment]
@ramiroJCB we would use the commit iterator in `gitutils` to retrieve the commits from a repo instead of what we're using now (`libgit2`)

**Comment by patrickdevivo** [comment]
Closing this out as our new container based sync approach makes it no longer relevant. We may or may not use gitutils in that approach in the future (today it continues to use `mergestat-lite`, but as a CLI)
